### DISCUSSION TITLE
Lots of room for optimisation building BVH

### DISCUSSION THREAD

**Discussion by Ono-Sendai** [jrouwe/JoltPhysics#369] [General]
Just had a quick profile of the Jolt shape building code. 
There's a lot of scope for making it faster.

For example on my test mesh with 122k vertices, 238k triangles,
Jolt takes about 133ms in JPH::AATreeBuilder::Build.

My BVH building code (running on a single thread, is usually multithreaded), takes 56ms.
So there's scope to make the Jolt code at least 2x faster.

For example, one thing you can do is just compute the triangle AABBs once, instead of recomputing them on this line:

bin.mBounds.Encapsulate(mVertices, GetTriangle(t));

You can recompute the centroid on the fly from the triangle AABBs efficiently.

Happy to discuss more,
  nick

**Comment by Ono-Sendai** [comment]
You can also take a look at the Embree code for peformance inspiration.  I do like the relative simplicity of the Jolt building code though!

**Comment by jrouwe** [comment]
Hello, there's an issue for the building speed here: https://github.com/jrouwe/JoltPhysics/issues/137

I'll look at your suggestions.
